paddle/framework/data_transform.cc

```diff
@@ -19,16 +19,19 @@ limitations under the License. */
 namespace paddle {
 namespace framework {
 
-Tensor* DataTransform(const OpKernelType& expected_kernel_type,
-                      const OpKernelType& kernel_type_for_var,
-                      const Tensor& input_tensor) {
-  Tensor* out = nullptr;
+void DataTransform(const OpKernelType& expected_kernel_type,
+                   const OpKernelType& kernel_type_for_var,
+                   const Tensor& input_tensor, Tensor* out) {
+  if (expected_kernel_type.library_type_ != kernel_type_for_var.library_type_) {
```

Comment:
out->set_library()?

Reply:
No, library is a mark of a kernel. Tensor does not have that field.